R/cube-transforms.R

```diff
@@ -0,0 +1,177 @@
+    includes <- c("subtotals")
+    if (headings) {
+        includes <- c(includes, "headings")
+    }
+    applyTransforms(x, include = includes)
+})
+
+#' Calculate the transforms from a CrunchCube
+#'
+#' `applyTransforms` calculates transforms (e.g. subtotals) on a CrunchCube.
+#' Currently only the row transforms are supported. This is useful if you want
+#' to use the values from the subtotals of the CrunchCube in an analysis.
+#'
+#' Including the `include` argument allows you to specify which parts of the
+#' CrunchCube to return. The options can be any of the following: "cube_cells"
+#' for the untransformed values from the cube itself, "subtotals" for the
+#' subtotal insertions, and "headings" for any additional headings. Any
+#' combination of these can be given, by default all will be given.
+#'
+#' `subtotalArray(cube)` is a convenience function that is equivalent to
+#' `applyTransforms(cube, include = c("subtotals"))`
+#'
+#' @param x a CrunchCube
+#' @param ary an array to use, if not using the default array from the cube
+#' itself. (Default: not used, pulls an array from the cube directly)
+#' @param ... arguments to pass to `calcTransforms`, for example `include`
+#' @param headings for `subtotalArray`: a logical indicating if the headings
+#' should be included with the subtotals (default: `FALSE`)
+#'
+#' @return an array with any transformations applied
+#'
+#' @examples
+#' \dontrun{
+#' transformed_array <- applyTransforms(crtabs(~opinion, ds))
+#'
+#' # to get an array of just the subtotals
+#' subtotalArray(crtabs(~opinion, ds))
+#' #    Agree Disagree 
+#' #       47       35 
+#' 
+#' # to get the full array with the subtotals but not headings
+#' applyTransforms(crtabs(~opinion, ds), include = c("cube_cells", "subtotals"))
+#' #             Strongly Agree             Somewhat Agree                      Agree 
+#' #                         23                         24                         47 
+#' # Neither Agree nor Disagree          Strongly Disagree                   Disagree 
+#' #                         18                         19                         35 
+#' #          Somewhat Disagree 
+#' #                         16 
+#' # to get the full array with the headings but not subtotals
+#' applyTransforms(crtabs(~opinion, ds), include = c("cube_cells", "headings")) 
+#' #               All opinions             Strongly Agree             Somewhat Agree 
+#' #                         NA                         23                         24 
+#' # Neither Agree nor Disagree          Strongly Disagree          Somewhat Disagree 
+#' #                         18                         19                         16 
+#' }
+#'
+#' @aliases subtotalArray
+#' @export
+applyTransforms <- function (x, ary = cubeToArray(x), ...) {
+    # if there are row transforms, calculate them and display them
+    row_trans <- tryCatch(Transforms(data=index(variables(x))[[1]]$view$transform), error = function(e) NULL)
```

Comment:
Why not use the (nearly identical) logic in https://github.com/Crunch-io/rcrunch/pull/139/files#diff-fc6c252cf7a4ec51064a3ca7fde28541R151 ? You're going to have to anyway when you implement applying transforms on all dims, not just 1.